File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

class Gigasecond {

    private LocalDate birthDate;
    private LocalDateTime birthDateTime;

    Gigasecond(LocalDate birthDate) {
        this.birthDate = birthDate;
    }

    Gigasecond(LocalDateTime birthDateTime) {
        this.birthDateTime = birthDateTime;
    }

    LocalDateTime getDate() {
        LocalDateTime current = this.birthDateTime;
        if (current == null) {
            current = this.birthDate.atStartOfDay();
        }
        int seconds = (int) Math.pow(10, 9);
        return current.plusSeconds(seconds);
    }

}

File: go/gigasecond/gigasecond.go
package gigasecond

// import path for the time package from the standard library
import (
	"math"
	"time"
)

const testVersion = 4

// AddGigasecond returns time when someone has lived for 10^9 seconds
func AddGigasecond(t time.Time) time.Time {
	tenToNine := int64(math.Pow(10, 9))
	seconds := time.Duration(tenToNine) * time.Second
	return t.Add(seconds)
}
